CMakeLists.txt

```diff
@@ -501,6 +510,7 @@ if(BUILD_SPARSE)
   file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR} SRCDIR)
   file(TO_NATIVE_PATH ${CMAKE_COMMAND} CMAKE_CMD)
   get_target_property(DGL_INCLUDE_DIRS dgl INCLUDE_DIRECTORIES)
+  string(REPLACE ";" "\\;" HIP_ARCHITECTURES_ESCAPED "${CMAKE_HIP_ARCHITECTURES}")
```

Comment:
Nit: I'd suggest just putting this somewhere towards the top. We have this in 2 conditions now, and it's not compute intensive.

Reply:
Done